src/data/sign.lean

```diff
@@ -0,0 +1,203 @@
+/-
+Copyright (c) 2022 Eric Rodriguez. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Rodriguez
+-/
+import order.basic
+import algebra.algebra.basic
+import tactic.derive_fintype
+
+/-!
+# Sign function
+
+This file defines the sign function for types with zero and a decidable less-than relation, and
+proves some basic theorems about it.
+-/
+
+/-- The type of signs. -/
+@[derive [decidable_eq, inhabited, fintype]]
+inductive sign_type
+| zero | neg | pos
+
+namespace sign_type
+
+instance : has_zero sign_type := ⟨zero⟩
+instance : has_one  sign_type := ⟨pos⟩
+
+instance : has_neg sign_type :=
+⟨λ s, match s with
+| neg  := pos
+| zero := zero
+| pos  := neg
+end⟩
+
+@[simp] lemma zero_eq_zero   : zero = 0 := rfl
+@[simp] lemma neg_eq_neg_one : neg = -1 := rfl
+@[simp] lemma pos_eq_one     : pos = 1  := rfl
```

Comment:
Do we get `-0 = 0` and `--1 = 1` via some other instance below?

Reply:
I just added `neg_zero'` in another PR, and so it's fine because of the distrib neg. We have neg_neg as we have an involutive negation